### PR TITLE
Misc improvements/v2

### PR DESCRIPTION
**Pull request by victorjulien**
Rebase #10016. Addresses comments. Adds a few more cleanups.

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 181 | 191 | 105.52% | 

Pipeline 17029

File: src/detect-engine-analyzer.c [hunk 1]
```diff
@@ -701,10 +701,10 @@ static void DumpContent(JsonBuilder *js, const DetectContentData *cd)
         jb_set_uint(js, "depth", cd->depth);
     }
     if (cd->flags & DETECT_CONTENT_DISTANCE) {
-        jb_set_uint(js, "distance", cd->distance);
+        jb_set_int(js, "distance", cd->distance);
```
Comment (catenacyber) [inline src/detect-engine-analyzer.c:704]: How come there were no warnings ?

Comment (catenacyber) [reply src/detect-engine-analyzer.c:704]: `Wimplicit-int-conversion` is only set in one build and this `DumpContent` is not always active ?

Comment (victorjulien) [reply src/detect-engine-analyzer.c:704]: No idea, the code compilation is not conditional.

**Comment by victorjulien** [comment]
replaced by #10052

File: src/detect-engine-analyzer.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-engine-analyzer.c:704]: https://redmine.openinfosecfoundation.org/issues/6632

Comment (catenacyber) [reply src/detect-engine-analyzer.c:704]: `Wsign-conversion` is the warning we want

**Pull request closed without merging**